### DISCUSSION TITLE
Application error: a client-side exception has occurred (see the browser console for more information).

### DISCUSSION THREAD

**Discussion by krushna06** [vercel/next.js#37375] [Help] [answered]

**Comment by icyJoseph** [comment, marked as answer]
Well, what did it say in the console? Do you have a link to the deployed site?

Does it also "work" if you run `npm run build` and then `npm run start` locally? It will show the production build on localhost:3000

I also cloned the repo and it works fine. Please do share a link to the Vercel deploy or try to copy what it says in the logs.

**Reply by krushna06** [reply]
Hi, thanks for the reply!
As I said, "it worked fine on localhost" which means my portfolio was running on localhost:3000 but the problem occurred on vercel. I thought your reply won't come so I joined next.js discord server and they asked me for the error. I had already deleted my site from vercel as I was getting the error but you asked me for the error and so did the next.js disocrd people, so I created the site again and magically it worked. 

I appreciate your quick response, thank you!

**Reply by tobiakinlade** [reply]
Hi Krushna06, I'm having the same issue on my deployed project to Vercel. Everything seems to be working fine on localhost but I'm getting this error ` Application error: a client-side exception has occurred (see the browser console for more information).` I deleted the project on Vercel and deployed another one, but it still same error.

**Reply by goburhan** [reply]
In my crash i am able to visit any page on desktop but when i trying to connect it with  mobile it gives me this error ....

**Reply by bis0072** [reply]
@goburhan i am also having the same issue.

**Reply by lack-of-gravitas** [reply]
Same issue here -- works on local. Get a flash of content then this extremely generic and unhelpful error.

**Reply by NitinSirsath** [reply]
Hey guys have you found the solution yet?
I have just encountered this error localhost has no issue, The problem comes with vercel only

**Reply by ekofi** [reply]
Same issue.

**Reply by NitinSirsath** [reply]
Hey! If you are using a private key for api call  then check console > network your request is working properly or getting undefined key, mine key is undefined so change environment variable in vercel deployment setting and issue got resolved.

**Reply by engageintellect** [reply]
Same issue, all of a sudden. Everything works locally. 

Thought it may be an issue with planetscale, but my db is up and working as expected.

**Reply by engageintellect** [reply]
**UPDATE:** It looks like Vercel didn't like the PlanetScale integration, so I removed it and set DATABASE_URL manually in Vercel's environment variables and all is good again.

You would think that enabling the PlanetScale integration would handle all of these things for you, but I guess not?

**Comment by daniel-akinyemi** [comment]
A client side error (from experience) on vercel usually occurs when you have not added your .local.env files in production.  (Make sure you have copied all your environment variables to your clip board) To do this go to your Settings -> Environment Variables -> Click on the 'Key' text field and paste the env's you copied -> Check production, preview and deployment -> Click save ->Go to your deployments and promote the latest (or a 'ready') to production. Hope this helps.
